components/courses/app/models/courses/lecture.rb

```diff
@@ -17,5 +17,11 @@ class Lecture < ApplicationRecord
     enum status: [UPCOMING, PASSED]
 
     validates :title, :started_at, :finished_at, :venue_id, :mentor_id, presence: true
+    validate  :time_policy
```

Comment:
Can you change `time_policy` name, because it validates that finished date should be after start data, and `time_policy` name does not say anything about that.
You can disagree with me of course.
Everything else looks good.